pages/index.tsx

```diff
@@ -2,20 +2,14 @@ import * as React from 'react'
 import AppNav from '../components/AppNav'
 import LandingPage from '../components/LandingPage'
 import Footer from '../components/Footer'
-import LoadingSpinner from '../components/LoadingSpinner'
-import { withGetApp, GetAppProps } from '../graphql'
 import { useRouter } from 'next/router'
-import _ from 'lodash'
-
-const IndexPage: React.FC<GetAppProps> = ({ data: { loading, session } }) => {
+import LoadingSpinner from '../components/LoadingSpinner'
+const IndexPage: React.FC<{}> = () => {
   const router = useRouter()
-
-  if (loading) return <LoadingSpinner />
-  if (session) {
+  if (process.browser && window.localStorage.getItem('loggedIn')) {
```

Comment:
More assured way to check if it's a browser.

```suggestion
  if (window && window.localStorage.getItem('loggedIn')) {
```

Reply:
This was my first approach but next throws errors `ReferenceError: window is not defined`. I'm not sure why it works this way now because I remember using this check in `AppNav` component for redirection. 
Alternatively I could use useEffect on mount([]) but process.browser should be a bit faster (it doesn't wait for a full load).

Reply:
Alright I see what you mean, but still `process` is not a standard object on browsers so it can confuse someone else that may work on this. There's a workaround you can use with `typeof` that won't cause that error because it doesn't evaluate `window`, only its type. See the suggestion below.

```suggestion
  if (
    typeof window !== 'undefined' &&
    window.localStorage.getItem('loggedIn')
  ) {
```

Besides that I think the `useEffect` would be more elegant and the difference in speed (if it exists) would be negligible since with this approach the redirection still only happens after the page is loaded, in both cases the JS redirection will only occur after the static HTML is rendered and the React scripts are downloaded and executed.

```suggestion
  useEffect(() => {
    localStorage.getItem('loggedIn') && router.push('/curriculum')
  }, [])
```

I did some quick tests here to compare total load times with redirection for these 3 cases. For the tests I disabled the cache and used the fast 3G network speed throttling. Then I went to the `/` location and waited until it redirected and the entire curriculum page loaded.

- With `useEffect`

  |Run|DOMContentLoaded|Load|Finish|
  |----|----------------|----|------|
  |1|1.04 s|9.91 s|10.03 s|
  |2|1.01 s|9.88 s|9.92 s|
  |3|1.02 s|9.85 s|10.01 s|

- With `if (typeof ...)` (without the loading spinner return)

  |Run|DOMContentLoaded|Load|Finish|
  |----|----------------|----|------|
  |1|1.02 s|9.86 s|9.95 s|
  |2|1.01 s|8.56 s|9.99 s|
  |3|1.01 s|8.59 s|9.98 s|

- With `if (process.browser ...)` (with the loading spinner return)

  |Run|DOMContentLoaded|Load|Finish|
  |----|----------------|----|------|
  |1|1.00 s|4.41 s|9.98 s|
  |2|1.02 s|4.43 s|10.02 s|
  |3|1.02 s|4.43 s|10.01 s|

As you can see the difference is not that significant and could be attributed to random variance. Also the DOMContentLoaded and Load times sometimes varied significantly in all cases, which in some runs the DOMContentLoaded even got below 1 s. The biggest difference is for the Load time in the last case, but that is due to returning the loading spinner, if I include the return in the second case the Load times are similar.

But still, the landing page flash after redirection is an issue we should try to find a solution, but it can be done later.

Reply:
OK, I changed it to `useEffect`.